--- a/src/corelib/thread/qyieldcpu.h
+++ b/src/corelib/thread/qyieldcpu.h
@@ -9,7 +9,11 @@
 #include <QtCore/qcompilerdetection.h>
 #include <QtCore/qprocessordetection.h>
 #include <QtCore/qtconfigmacros.h>
 
+#if defined(__APPLE__) && (defined(__ARM_ARCH) || defined(__aarch64__) || defined(__arm__))
+#  include <arm_acle.h>
+#endif
+
 #ifdef Q_CC_MSVC_ONLY
 // MSVC defines _YIELD_PROCESSOR() in <xatomic.h>, but as that is a private
 // header, we include the public ones
